pkg/osvscanner/vulnerability_result.go

```diff
@@ -110,5 +110,6 @@ func buildVulnerabilityResults(
 		results.ExperimentalAnalysisConfig.Licenses.Allowlist = allowlist
 	}
 
+	results.ResultsByPURL = grouper.GroupByPURL(results.Results)
```

Comment:
Can we avoid storing this, and just compute these on demand when this is needed?